docs/how-to-guides/meson-args.rst

```diff
@@ -17,5 +16,98 @@ Passing arguments to Meson_
    - Mention the order in which Meson arguments are added, and how it affect
      Meson
 
+Advanced meson options can be accessed by specifying extra arguments to the individual meson comands:
+
+   - meson dist: https://mesonbuild.com/Commands.html#dist
+   - meson setup: https://mesonbuild.com/Commands.html#setup
+   - meson compile: https://mesonbuild.com/Commands.html#compile
+   - meson install: https://mesonbuild.com/Commands.html#install
+
+These arguments can be added to the project via the project's ``pyproject.toml``:
+
+.. code-block:: console
+
+   [tool.meson-python.args]
+   dist = ['dist-argument_1','dist-argument_2', '...']]
+   setup = ['setup-argument_1','setup-argument_2', '...']]
+   compile = ['compile-argument_1','compile-argument_2', '...']]
+   install = ['install-argument_1','install-argument_2', '...']]
+
+.. admonition:: All four categories must exist
+   :class: attention
+
+   Currently all four option categories (dist, setup, compile and install) must be present in the [tool.meson-python.args] section, even if no arguments are supplied (empty list: [])
+
+Or using the build config settings:
+
+.. code-block:: console
+
+   # with pypa/build
+   python -m build -Cdist-args="args"    \
+                   -Csetup-args="args"   \
+                   -Ccompile-args="args" \
+                   -Cinstall-args="args" .
+
+   # with pip
+   python -m pip --config-settings=dist-args="args"    \
+                 --config-settings=setup-args="args"   \
+                 --config-settings=compile-args="args" \
+                 --config-settings=install-args="args" .
```

Comment:
Please split the commands into separate code-blocks, and add a `$` at the start (it gets ignored when selecting it).